app/models/concerns/has_slug.rb

```diff
@@ -5,9 +5,12 @@ module HasSlug
     extend ActiveSupport::Concern
 
     included do
-      validates :slug, length: {maximum: 64, minimum: 3}, uniqueness: true
+      validates :slug, format: { with: /\A[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*\z/,
+                                 message: I18n.t('errors.service.slug.invalid')},
```

Comment:
I think the following regex will do what you want, and is a bit shorter: `\A[a-z0-9][a-z0-9.-]+[a-z0-9]\z`

Reply:
By the way, if you don't know it already, www.rubular.com is a great resource for playing around with, and testing, regular expressions.

Reply:
As I understand it, that regex will allow

`foo--------bar`

Is there any reason for us to allow multiple dashes? Not technical, but I wouldn't want to have to distinguish between `foo---bar` and `foo----bar` etc.  If we don't then the regex could be altered to 

`\A([a-z0-9]+-)*[a-z0-9]+\z`

Reply:
The regex is used by Kubernetes, it's not something I generated myself. Personally I can't see a reason to allow multiple dashes, so happy to go with your regex expression

Reply:
@theKHutDeveloper Yours does just as well. I was merely refining @stephenrichards 's refinement ;)

Reply:
@solidgoldpig, @stephenrichards: The regex `\A([a-z0-9]+-)*[a-z0-9]+\z` does not allow full stops such as **example.com** which is something that Kubernetes allows in their regex. I have amended the initial regex to also allow only one '-' between text chars `\A[a-z0-9]([-a-z0-9]?[a-z0-9])*(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*\z`.